### ISSUE TITLE
Overwrite the SSH user in the verify step

### ISSUE THREAD

**Issue by doertedev** [inspec/kitchen-inspec#231]
### Describe the Enhancement:
TL;DR: Verify a connection via a freshly created user.

Longer version:
I throw away the distro's default user my cloud provider wants me to connect as when initially connecting to the instance. Got all of that wrapped in chef, creating a new default user to SSH against, which would mean that i need to connect to the instance with different credentials in the verify step as in the converge step. Otherwise I'm stuck in an infinite loop when hitting `kitchen verify`.

### Describe the Need:
Anyone changing SSHD and/or user configuration during the provisioning step.

### Current Alternative
Skip using kitchen and having to execute chef + inspec on an instance via ssh.

### Can We Help You Implement This?:
This would be another config[] || kitchen[] right? (https://github.com/inspec/kitchen-inspec/blob/master/lib/kitchen/verifier/inspec.rb#L233)

**Comment by clintoncwolfe** [comment]
Hi! Great suggestion.  However, I don't think this is specific  to kitchen-inspec - the change would need to be made in Test Kitchen, so that any verifier could use the new feature. Please open this issue in the test-kitchen repo - https://github.com/test-kitchen/test-kitchen 

Thanks!
